src/carma-1-tenth/plot_crosstrack_error.py

```diff
@@ -28,17 +30,28 @@ def is_left(route_a, route_b, odometry):
         return True
     else:
         return False
+    
+def get_route_coordinates(route_message):
+    route_coordinates = []
+    # Rotate the route_graph coordinates 90 degrees to match C1T coordinates (x-forward, y-left)
+    if type(route_message) == MarkerArray:
+        for i in range(len(route_message.markers)):
+            if route_message.markers[i].type == 2:
+                route_coordinates.append([-route_message.markers[i].pose.position.y, route_message.markers[i].pose.position.x])
+    elif type(route_message) == Path:
+        for i in range(len(route_message.poses)):
+            route_coordinates.append([-route_message.poses[i].pose.position.y, route_message.poses[i].pose.position.x])
+    return route_coordinates
 
 
-def plot_crosstrack_error(bag_dir, show_plots=True):
+def plot_crosstrack_error(bag_dir, route_topic, show_plots=True):
     metadatafile : str = os.path.join(bag_dir, "metadata.yaml")
     if not os.path.isfile(metadatafile):
         raise ValueError("Metadata file %s does not exist. Are you sure %s is a valid rosbag?" % (metadatafile, bag_dir))
     with open(metadatafile, "r") as f:
         metadata_dict : dict = yaml.load(f, Loader=yaml.SafeLoader)["rosbag2_bagfile_information"]
     storage_id = metadata_dict['storage_identifier']
     odom_topic = '/amcl_pose'
```

Comment:
at face value this assignment is a bit confusing even if it is old code. Not sure if there's anything to do about it to differentiate between relative vs absolute odometry here but figured it was worth mentioning

Reply:
Added a comment to clarify this is odom in map frame